File: python/easy/binary_search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums) - 1
        while l <= r:
            p = (r+l)//2
            if nums[p] == target:
                return p
            elif nums[p] > target:
                r = p -1
            else:
                l = p + 1
        return -1

File: go/easy/binary-search.go
func search(nums []int, target int) int {
	return binarySearch(nums[:], 0, len(nums), target)
}

func binarySearch(nums []int, l int, r int, target int) int {
	if l >= r {
		return -1
	}

	m := l + (r-l)/2
	if nums[m] > target {
		return binarySearch(nums[:], l, m, target)
	} else if nums[m] < target {
		return binarySearch(nums[:], m+1, r, target)
	} else {
		return m
	}
}
